ql/src/java/org/apache/hadoop/hive/ql/exec/tez/CustomPartitionVertex.java

```diff
@@ -221,28 +211,26 @@ public void onRootVertexInitialized(String inputName, InputDescriptor inputDescr
       } else if (event instanceof InputDataInformationEvent) {
         dataInformationEventSeen = true;
         InputDataInformationEvent diEvent = (InputDataInformationEvent) event;
-        FileSplit fileSplit;
+        HiveInputSplit inputSplit;
         try {
-          fileSplit = getFileSplitFromEvent(diEvent);
+          inputSplit = getInputSplitFromEvent(diEvent);
         } catch (IOException e) {
           throw new RuntimeException("Failed to get file split for event: " + diEvent, e);
         }
-        Set<FileSplit> fsList =
-            pathFileSplitsMap.get(Utilities.getBucketFileNameFromPathSubString(fileSplit.getPath()
-                .getName()));
-        if (fsList == null) {
-          fsList = new TreeSet<FileSplit>(new PathComparatorForSplit());
-          pathFileSplitsMap.put(
-              Utilities.getBucketFileNameFromPathSubString(fileSplit.getPath().getName()), fsList);
+        final int bucketId = inputSplit.getBucketId().orElse(-1);
+        Set<HiveInputSplit> inputSplits = bucketFileSplitsMap.get(bucketId);
+        if (inputSplits == null) {
```

Comment:
i think you can use `Map.computeIfAbsent` instead

Reply:
Done
https://github.com/apache/hive/pull/5409/commits/ae7444e8e1c2de95bfc1915af3d3c0245af4b472